### PR TITLE
Clean up our github issue templates

### LINKED ISSUE #2661: Make our issue templates less scary :-p
There are [reports](https://twitter.com/wycats/status/1362154625869123584) that our issue templates are intimidating

### PR DESCRIPTION
**Pull request by fitzgen**
* Don't use HTML comments, as they are noisy, which makes the templates more
  intimidating.

* Use "TODO" to clearly demarcate everywhere the issue reporter should fill in
  some information.

* Use headers and white space abundantly, which makes it easier to visually
  process the template and its sections at a glance, and less of an intimidating
  wall of text.

Fixes #2661

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/wasmtime-bug-report.md [hunk 1]
```diff
@@ -1,18 +1,40 @@
 ---
-name: Wasmtime bug report
+name: Wasmtime Bug Report
```
Comment (bjorn3) [inline .github/ISSUE_TEMPLATE/wasmtime-bug-report.md:2]: There is still some inconsistency in casing between templates.

**Review by pchickey — APPROVED** [review]
Thank you!

File: .github/ISSUE_TEMPLATE/clif-bug-report.md [hunk 2]
```diff
@@ -7,13 +7,37 @@ assignees: ''
...
+
+TODO: What actually happens? Panic? Segfault? Incorrect result?
+
+### Versions and Environment
+
+Cranelift version or commit: TODO
+
+Operating system: TODO
+
+Architecture: TODO
```
Comment (abrown) [inline .github/ISSUE_TEMPLATE/clif-bug-report.md:39]:
```suggestion
Cranelift version or commit: e.g., `git describe`?

Operating system: e.g., `uname -osr`

Architecture: e.g., `lscpu | egrep 'Architecture|Model name'`
```

**Review by abrown — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/clif-bug-report.md (hunk 2 above)
Comment (abrown) [reply .github/ISSUE_TEMPLATE/clif-bug-report.md:39]: I mean, we might want to think up the Windows/MacOS versions of some of those but that would answer @wycats' comments [here](https://github.com/bytecodealliance/wasmtime/issues/2661#issuecomment-780875992).

Comment (fitzgen) [reply .github/ISSUE_TEMPLATE/clif-bug-report.md:39]: This removes the TODO hints that every other field they need to fill out has, makes the template wordier (which makes it more like an intimidating wall of text), and assumes a unix environment. Not convinced that this change is a clear win.

Comment (cfallin) [reply .github/ISSUE_TEMPLATE/clif-bug-report.md:39]: +1 -- OS and architecture would likely be known by folks who would be reporting issues (we can always have a conversation in an issue if not or if we need more detail); whereas complex command pipelines make it look like we're asking for very specific and picky details and may scare folks away, I think.

**Pull request merged**